### DISCUSSION TITLE
How to implement Sign in with Apple

### DISCUSSION THREAD

**Discussion by saeedesmaili** [fastapi-users/fastapi-users#1321] [Q&A]
Has anyone seen a comprehensive guide on how to add "Sign in with Apple" using fastapi-users? As far as I understand, Apple uses OpenID so I should be able to use [httpx-oauth](https://frankie567.github.io/httpx-oauth/oauth2/#openid) (and fastapi-users), but there is no mention of apple in these two libraries' docs and Apple's documentation is also very confusing. I have implemented oauth2 with google and a few others, but I have zero experience with apple's dev experience.

Here is what I've found on apple's docs, but I'm even struggling to find where should I look for `CLIENT_ID`, `CLIENT_SECRET`, and `the OpenID configuration endpoint` to use with httpx-oauth.
https://developer.apple.com/documentation/sign_in_with_apple/sign_in_with_apple_rest_api/authenticating_users_with_sign_in_with_apple

**Comment by saeedesmaili** [comment]
Ah, I probably need to enroll to apple's developer program first!

**Comment by frankie567** [comment]
I don't have experience with Apple OAuth, but yes, if they implement the OpenID protocol, it should work with the generic OpenID client. Keep us tuned!

**Reply by frankie567** [reply]
Here is the OpenID configuration endpoint: https://appleid.apple.com/.well-known/openid-configuration

And yes, you need to have a developer account and create an OAuth app on their side to get your own client ID and secret!
